code/modules/reagents/chemistry/reagents/food_reagents.dm

```diff
@@ -155,9 +155,14 @@
 	taste_mult = 1.5 // stop sugar drowning out other flavours
 	nutriment_factor = 10 * REAGENTS_METABOLISM
 	metabolization_rate = 2 * REAGENTS_METABOLISM
-	overdose_threshold = 200 // Hyperglycaemic shock
+	overdose_threshold = 75 // Hyperglycaemic shock
 	taste_description = "sweetness"
 
+/datum/reagent/consumable/sugar/on_mob_life(mob/living/carbon/C)
+	if(C.has_trait(TRAIT_DIABETIC))
+		/datum/reagent/consumable/sugar
+		overdose_threshold = 15
```

Comment:
Why is this on life and not say on add?

Reply:
because I am new to coding and am likely to not know better solutions, thanks for telling me about that

Reply:
lol'd